src/main/kotlin/de/bund/digitalservice/useid/apikeys/ApiKeyAuthenticationFilter.kt

```diff
@@ -0,0 +1,50 @@
+package de.bund.digitalservice.useid.apikeys
+
+import org.springframework.http.HttpHeaders
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.web.filter.OncePerRequestFilter
+import javax.servlet.FilterChain
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+private const val AUTH_HEADER_VALUE_PREFIX = "Bearer "
+
+class ApiKeyAuthenticationFilter(private val authenticationManager: AuthenticationManager) :
+    OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        val authHeader: String? = extractAuthHeader(request)
+        val token = authHeader?.substring(AUTH_HEADER_VALUE_PREFIX.length)
```

Comment:
```suggestion
        val key = authHeader?.substring(AUTH_HEADER_VALUE_PREFIX.length)
```

I think it is cleaner to use key, because we read the key in this line and the following line with `ApiKeyAuthenticationToken`creates a token.

Reply:
```suggestion
        val apiKey = authHeader?.substring(AUTH_HEADER_VALUE_PREFIX.length)
```
I agree. What about apiKey to make it more explicit?

Reply:
Done